### PR TITLE
feat(plonky2): Added plonky2_verify_failure and plonky2_verify_success tests

### PR DESCRIPTION
**Pull request by nickysn**

### CODE REVIEW

File: test_programs/plonky2_verify_failure/zk_dungeon_verify_fail_1/src/tests.nr
Comment (smanilov) [inline test_programs/plonky2_verify_failure/zk_dungeon_verify_fail_1/src/tests.nr:1]: Same here: I don't think we need this file.

File: test_programs/plonky2_verify_failure/zk_dungeon_verify_fail_1/files-to-test.json
Comment (smanilov) [inline test_programs/plonky2_verify_failure/zk_dungeon_verify_fail_1/files-to-test.json:1]: Same here: not ran with `cargo test` so I think we don't need it. (But maybe I'm wrong.)

File: test_programs/plonky2_verify_failure/zk_dungeon_verify_fail_1/VerifierTest.toml
Comment (smanilov) [inline test_programs/plonky2_verify_failure/zk_dungeon_verify_fail_1/VerifierTest.toml:1]: What's with the file name? Shouldn't this be just Verifier.toml?

File: test_programs/plonky2_verify_failure/zk_dungeon_verify_fail_1/package.json
Comment (smanilov) [inline test_programs/plonky2_verify_failure/zk_dungeon_verify_fail_1/package.json:1]: Same here: I don't think we need this file.

File: test_programs/plonky2_verify_failure/zk_dungeon_verify_fail_1/src/tests/valid_path_test_1.nr
Comment (smanilov) [inline test_programs/plonky2_verify_failure/zk_dungeon_verify_fail_1/src/tests/valid_path_test_1.nr:1]: Same here: I don't think we need this directory.

File: test_programs/plonky2_verify_failure/zk_dungeon_verify_fail_1/test/IsSafePath.test.js
Comment (smanilov) [inline test_programs/plonky2_verify_failure/zk_dungeon_verify_fail_1/test/IsSafePath.test.js:1]: Same here: I don't think we need this directory.

File: test_programs/plonky2_verify_failure/zk_dungeon_verify_fail_1/VerifierTest.toml
Comment (smanilov) [reply test_programs/plonky2_verify_failure/zk_dungeon_verify_fail_1/VerifierTest.toml:1]: Oh, I see, `nargo prove` generates Verifier.toml, so you don't want this to be overwritten.

Perhaps add a comment inside the file?

File: tooling/nargo_cli/build.rs
Comment (smanilov) [inline tooling/nargo_cli/build.rs:757]: You could use this to check the exact error message.

It would be ("zk_dungeon_verify_fail_1", vec!["VerificationFailed { message: \"Public inputs don't match proof\" }."])

Alternatively, you could just remove this and related mechanism for checking messages match.

**Review by smanilov — COMMENTED** [review]
Please, take a look at the comments before merging.

Do we need all the JS tests? They seem to be testing the zk_dungeon program, rather than testing the verify command, so I think they are actually noise. Also, they don't seem to be running automatically with `cargo test`, so they are unlikely to be used. Or am I missing something.

File: test_programs/plonky2_verify_failure/zk_dungeon_verify_fail_1/files-to-test.json
Comment (nickysn) [reply test_programs/plonky2_verify_failure/zk_dungeon_verify_fail_1/files-to-test.json:1]: Removed.

File: test_programs/plonky2_verify_failure/zk_dungeon_verify_fail_1/package.json
Comment (nickysn) [reply test_programs/plonky2_verify_failure/zk_dungeon_verify_fail_1/package.json:1]: Removed.

File: test_programs/plonky2_verify_failure/zk_dungeon_verify_fail_1/src/tests/valid_path_test_1.nr
Comment (nickysn) [reply test_programs/plonky2_verify_failure/zk_dungeon_verify_fail_1/src/tests/valid_path_test_1.nr:1]: Removed.

File: test_programs/plonky2_verify_failure/zk_dungeon_verify_fail_1/test/IsSafePath.test.js
Comment (nickysn) [reply test_programs/plonky2_verify_failure/zk_dungeon_verify_fail_1/test/IsSafePath.test.js:1]: Removed.

File: test_programs/plonky2_verify_failure/zk_dungeon_verify_fail_1/VerifierTest.toml
Comment (nickysn) [reply test_programs/plonky2_verify_failure/zk_dungeon_verify_fail_1/VerifierTest.toml:1]: Done.

**Review by smanilov — APPROVED** [review]
LGTM, but please address remaining comment.

File: tooling/nargo_cli/build.rs
Comment (nickysn) [reply tooling/nargo_cli/build.rs]: Done. Unfortunately untested, because the tests fail, due to upstream breakage of the verify command.

**Comment by nickysn** [comment]
> LGTM, but please address remaining comment.

It has been addressed, however merging this will cause the tests to fail, because 'verify' is broken by the changes made upstream.

**Comment by nickysn** [comment]
This can be merged after: https://github.com/blocksense-network/noir/pull/9

**Comment by smanilov** [comment]
> > LGTM, but please address remaining comment.
> 
> It has been addressed, however merging this will cause the tests to fail, because 'verify' is broken by the changes made upstream.

This would have been fine, even better, I think. Because then it would be easier to check that the other PR indeed fixes the tests. Not that important, though, so I'm merging the PRs in the order you anticipated.

**Pull request merged**
